File: services/backend/src/core_service/modules/devices/domain/device.py
"""Device 도메인 엔티티 — schema.md §5 `devices` 테이블 매핑.

설치모드 판별 임계값(decisions.md #5, 확정 — 임의 변경 금지):
  RAM < 6GB 또는 Android <= 11 (API 30) → kiosk, 그 외 normal.
"""

from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal
from enum import StrEnum
from uuid import UUID

KIOSK_RAM_THRESHOLD_GB = Decimal("6.0")
KIOSK_ANDROID_MAX_VERSION = 11


class InstallMode(StrEnum):
    KIOSK = "kiosk"
    NORMAL = "normal"


def determine_install_mode(ram_gb: Decimal, android_version: str) -> InstallMode:
    """decisions.md #5 — 보수적 판정(둘 중 하나라도 미달 시 키오스크).

    ⚠️ android_version은 문자열(예: "9", "11", "13")로 저장되므로 사전식 비교(<=)를
    쓰면 "9" <= "11"이 False로 잘못 평가된다(문자 '9' > '1'). 반드시 정수로 변환해
    비교한다 — 메이저 버전만 온다는 전제(마이너 버전 포함 시 파싱 로직 확장 필요).
    """
    major_version = int(android_version.split(".")[0])
    if ram_gb < KIOSK_RAM_THRESHOLD_GB or major_version <= KIOSK_ANDROID_MAX_VERSION:
        return InstallMode.KIOSK
    return InstallMode.NORMAL


@dataclass
class Device:
    id: UUID
    display_id: str
    model_name: str | None
    user_id: UUID
    ram_gb: Decimal
    android_version: str
    install_mode: InstallMode
    ai_tops: Decimal | None
    slm_model_version: str | None  # schema.md v1.3, decisions.md #35
    prompt_pack_version: str | None  # schema.md v1.3, decisions.md #35
    installed_at: datetime
    last_sync_at: datetime | None

File: apps/admin/src/types/device.ts
/** design.md §3.1(v0.7) Device — schema.md `devices` 테이블과 1:1 매핑.
 * apps/web/src/types/device.ts와 동일 — 공유 패키지가 아직 없어 두 앱이 각자 들고 있다. */
export type InstallMode = "kiosk" | "normal";

export interface Device {
  id: string;
  displayId: string;
  modelName?: string;
  userId: string;
  ramGb: number;
  androidVersion: string;
  installMode: InstallMode;
  aiTops?: number;
  slmModelVersion?: string;
  promptPackVersion?: string;
  installedAt: string;
  lastSyncAt?: string;
}

/** 한글 표시명 — schema.md §7에는 아직 install_mode 매핑이 없어(devices/sync_status
 * 계열 표는 §7에 없음) apps/admin 스캐폴딩 시 새로 정한 값. 나중에 §7에 정식 편입 필요. */
export const INSTALL_MODE_LABEL: Record<InstallMode, string> = {
  kiosk: "키오스크",
  normal: "일반",
};
